c10/macros/Macros.h

```diff
@@ -372,6 +373,14 @@ __host__ __device__
                static_cast<unsigned>(__LINE__)), \
            0);                                   \
   }
+#define CUDA_KERNEL_ASSERT_MSG(cond, msg)        \
+  if (C10_UNLIKELY(!(cond))) {                   \
+    (void)(_wassert(                             \
+               _CRT_WIDE(msg),                   \
```

Comment:
I think you can only apply this to a string, not variable. Hence the windows build failure.

Reply:
>:(

Reply:
I'm just going to not throw the device-side assert on windows. A brief search did not show up any easy ways of converting a char* to a wchar*.